Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -539,6 +574,204 @@ void GridTrackSizingAlgorithm::increaseSizesToAccommodateSpanningItems(const Gri
+        }
+    };
+
+    StdSet<unsigned> spanLengths;
+    WTF::StdMultimap<unsigned, MasonryMinMaxTrackSizeWithGridSpan> multiTrackItems;
+    auto& allTracks = tracks(m_direction);
+
+    for (auto& gridTrackSpan : gridTrackSpans) {
+        for (auto trackIndex = 0u; trackIndex < allTracks.size(); trackIndex++) {
+            auto endLine = trackIndex + gridTrackSpan.first;
+            auto itemSpan = GridSpan::translatedDefiniteGridSpan(trackIndex, endLine);
+
+            if (itemSpan.integerSpan() == 1)
+                continue;
+
+            if (endLine > allTracks.size())
+                continue;
+
+            if (spanningItemCrossesFlexibleSizedTracks(itemSpan))
+                continue;
+
+            multiTrackItems.insert({ itemSpan.integerSpan(), MasonryMinMaxTrackSizeWithGridSpan { .trackSize = gridTrackSpan.second, .gridSpan = itemSpan } });
+
+            if (!spanLengths.contains(itemSpan.integerSpan()))
+                spanLengths.insert(itemSpan.integerSpan());
+        }
+    }
+
+    for (auto& multiTrackItem : definiteMultiTrackItems) {
+        auto& gridItem = multiTrackItem.second.gridItem();
+        auto minContentForGridItem = m_strategy->minContentForGridItem(gridItem, gridLayoutState);
+        auto maxContentForGridItem = m_strategy->maxContentForGridItem(gridItem, gridLayoutState);
+        auto minSizeForGridItem = m_strategy->minSizeForGridItem(gridItem, gridLayoutState);
+
+        multiTrackItems.insert({ multiTrackItem.first, MasonryMinMaxTrackSizeWithGridSpan { .trackSize = MasonryMinMaxTrackSize { .minContentSize = minContentForGridItem, .maxContentSize = maxContentForGridItem, .minSize = minSizeForGridItem }, .gridSpan = multiTrackItem.second.span() } });
+
+        if (!spanLengths.contains(multiTrackItem.first))
+            spanLengths.insert(multiTrackItem.first);
+    }
+
+    for (auto span : spanLengths) {
+        increaseSizes.template operator()<TrackSizeComputationPhase::ResolveIntrinsicMinimums>(span, multiTrackItems);
+        increaseSizes.template operator()<TrackSizeComputationPhase::ResolveContentBasedMinimums>(span, multiTrackItems);
+        increaseSizes.template operator()<TrackSizeComputationPhase::ResolveMaxContentMinimums>(span, multiTrackItems);
+        increaseSizes.template operator()<TrackSizeComputationPhase::ResolveIntrinsicMaximums>(span, multiTrackItems);
+        increaseSizes.template operator()<TrackSizeComputationPhase::ResolveMaxContentMaximums>(span, multiTrackItems);
+    }
+
+}
+
+template <TrackSizeComputationVariant variant>
+void GridTrackSizingAlgorithm::masonryIncreaseSizesToAccommodateSpanningItemsWithFlex(const StdMap<unsigned, MasonryMinMaxTrackSize> gridTrackSpans, WTF::StdMultimap<unsigned, GridItemWithSpan> definiteMultiTrackItems, GridLayoutState& gridLayoutState)
+{
+    auto increaseSizes = [&]<TrackSizeComputationPhase phase>(StdSet<unsigned> spans, WTF::StdMultimap<unsigned, MasonryMinMaxTrackSizeWithGridSpan> multiTrackItems)
```

Comment:
Same thing here let's move it outside this method definition

Reply:
Removed most of the other code out of this function. Please re-evaluate.